pkg/dao/rpms.go

```diff
@@ -210,9 +211,72 @@ func (r rpmDaoImpl) Search(ctx context.Context, orgID string, request api.Conten
 		return nil, db.Error
 	}
 
+	// Add module info to the response if requested
+	if request.IncludePackageSources && len(dataResponse) > 0 {
+		err := r.addModuleInfo(ctx, dataResponse, repoUUIDs)
+		if err != nil {
+			return nil, err
+		}
+	}
+
 	return dataResponse, nil
 }
 
+func (r *rpmDaoImpl) addModuleInfo(ctx context.Context, rpmResponse []api.SearchRpmResponse, repoUUIDs []string) error {
+	var moduleInfo []models.ModuleStream
+
+	err := r.db.WithContext(ctx).
+		Table("module_streams").
+		Joins("inner join repositories_module_streams rms ON rms.module_stream_uuid = module_streams.uuid").
+		Select("name", "stream", "context", "arch", "version", "description", "package_names").
+		Where("rms.repository_uuid in (?)", repoUUIDs).
+		Find(&moduleInfo).Error
+	if err != nil {
+		return err
+	}
+
+	moduleMap := make(map[string][]api.ModuleInfoResponse)
+	for _, m := range moduleInfo {
+		moduleMap[m.Name] = append(moduleMap[m.Name], api.ModuleInfoResponse{
+			Type:        "module",
+			Name:        m.Name,
+			Stream:      m.Stream,
+			Context:     m.Context,
+			Arch:        m.Arch,
+			Version:     m.Version,
+			Description: m.Description,
+		})
+	}
+
+	for i, rpm := range rpmResponse {
+		var matchedModules []api.ModuleInfoResponse
+		for _, m := range moduleInfo {
+			if m.Name == rpm.PackageName || utils.Contains(m.PackageNames, rpm.PackageName) {
```

Comment:
I'm not sure we need to check here if `m.Name == rpm.PackageName`  while sometimes they will match, the package name should always be in the list.  Curious if there was a data reason for that check

Reply:
oh good point! i had this check initially before realizing i had a bug and decided to also/instead check the package list :) i don't think we need this, let me remove